Replaces `this.get with ES5 getter where possible (#2303) in v3.2.0

cspanring opened the pull request:
Attempt to check off task 1 in https://github.com/emberjs/guides/issues/2303

@jenweber please have a look and/or pull in more eyes to make sure the changes are OK.

jenweber commented on guides/v3.2.0/components/the-component-lifecycle.md:
```diff
@@ -66,7 +66,7 @@ import Component from '@ember/component';
   init() {
     this._super(...arguments);
-    this.errors = [];
+    this.set('errors', []);
```
This is a really good catch, thank you

mansona commented on guides/v3.2.0/models/relationships.md:
```diff
@@ -190,7 +190,7 @@ export default PaymentMethod.extend({
 
   obfuscatedIdentifier: computed('linkedEmail', function () {
-    let last5 = this.get('linkedEmail').split('').reverse().slice(0, 5).reverse().join('');
+    let last5 = this.linkedEmail.split('').reverse().slice(0, 5).reverse().join('');
```
does this work with Models? I have in the back of my mind that you still need to use `this.get()` for some things and I have a feeling that it has something to do with Ember Data but I don't know the specifics... Can anyone confirm?

cspanring replied:
Will verify. ES5 getters can't be used for anything that implements a `Proxy` and Ember Data is using them a lot.

mansona commented on guides/v3.2.0/object-model/classes-and-instances.md:
```diff
@@ -117,7 +117,7 @@ import EmberObject from '@ember/object';
 const Person = EmberObject.extend({
   helloWorld() {
-    alert(`Hi, my name is ${this.get('name')}`);
+    alert(`Hi, my name is ${tthis.name}`);
```
double `t` in `tthis`

cspanring replied:
good catch!

mansona commented on guides/v3.2.0/object-model/classes-and-instances.md:
```diff
@@ -152,7 +152,7 @@ import EmberObject from '@ember/object';
 const Person = EmberObject.extend({
   init() {
-    alert(`${this.get('name')}, reporting for duty!`);
+    alert(`${tthis.name}, reporting for duty!`);
```
ttypo 😂

CodingItWrong commented on guides/v3.2.0/components/handling-events.md:
```diff
@@ -69,7 +69,7 @@ export default Component.extend({
   drop(event) {
     let id = event.dataTransfer.getData('text/data');
-    this.get('action')(id);
+    this.action(id);
```
I'd second this. I could see myself getting confused by an action called "action"

cspanring replied:
agree, will do.

CodingItWrong commented on guides/v3.2.0/components/the-component-lifecycle.md:
```diff
@@ -237,8 +237,8 @@ export default Component.extend({
-    this.set('items', this.get('items').map((item) => {
-      if (item.id === this.get('selectedItem.id')) {
+    this.set('items', this.items.map((item) => {
+      if (item.id === this.selectedItem.id) {
```
Unrelated to this PR, but it seems like this `map()` is mutating the items. I think some kind of for…of or forEach could better express that we are changing the items in place. But no need to make the change in this PR unless you feel especially motivated to do so :-)

cspanring replied:
I can look at it and propose a better example, no problem.

CodingItWrong commented on guides/v3.2.0/components/triggering-changes-with-actions.md:
```diff
@@ -186,7 +186,7 @@ export default Component.extend({
 
-`this.get('onConfirm')` will return the function passed from the parent as the
+`this.onConfirm` will return the function passed from the parent as the
 value of `onConfirm`, and the following `()` will invoke the function.
```
I wonder if we still need the explanation about "the following ()". When "get()" was needed I _did_ feel like I needed the explanation of the extra parens, but now it's just invoking a function property as usual.

cspanring replied:
yep, agree, can tweak or remove that phrase.

CodingItWrong commented on guides/v3.2.0/object-model/computed-properties-and-aggregate-data.md:
```diff
@@ -25,7 +25,7 @@ export default Component.extend({
 
   titles: computed('todos.[]', function() {
-    let todos = this.get('todos');
+    let todos = this.todos;
```
Could probably just inline `this.todos` in the following line, unless you think there's extra clarity separating it out.

CodingItWrong commented on guides/v3.2.0/object-model/computed-properties.md:
```diff
@@ -20,8 +20,8 @@ Person = EmberObject.extend({
-    let firstName = this.get('firstName');
-    let lastName = this.get('lastName');
+    let firstName = this.firstName;
+    let lastName = this.lastName;
```
I'd probably in-line these into the template string instead of assigning to local variables

CodingItWrong commented on guides/v3.2.0/object-model/computed-properties.md:
```diff
@@ -57,8 +57,8 @@ import Ember from 'ember':
-    let firstName = this.get('firstName');
-    let lastName = this.get('lastName');
+    let firstName = this.firstName;
+    let lastName = this.lastName;
```
Same as previous

CodingItWrong commented on guides/v3.2.0/testing/testing-components.md:
```diff
@@ -15,7 +15,7 @@ export default Component.extend({
 
   style: computed('name', function() {
-    const name = this.get('name');
+    const name = this.name;
```
Same as above

CodingItWrong commented on guides/v3.2.0/tutorial/autocomplete-component.md:
```diff
@@ -229,14 +229,14 @@ export default Controller.extend({
-        return this.get('store').query('rental', { city: param });
-        return this.get('store')
+        return this.store.query('rental', { city: param });
+        return this.store
```
It might just be the formatting here on my phone, but I'm not even sure I understand what's happening in these lines. Is it two return statements in a row? The second one would never be reached, right? If so I think it'd be good to quickly clean it up as part of this PR.

cspanring replied:
good catch, this needs to be fixed, will do.

cspanring replied:
this is displayed as diff in the rendered version: https://guides.emberjs.com/release/tutorial/autocomplete-component/#toc_handling-results-coming-back-at-different-times
the redundant code makes sense here.

CodingItWrong commented on guides/v3.2.0/tutorial/autocomplete-component.md:
```diff
@@ -229,14 +229,14 @@ export default Controller.extend({
-        return this.get('store').findAll('rental');
-        return this.get('store')
+        return this.store.findAll('rental');
+        return this.store
```
Same as above

cspanring replied:
same, displayed as diff in the rendered guides.

CodingItWrong commented on guides/v3.2.0/tutorial/service.md:
```diff
@@ -196,14 +196,14 @@ export default Component.extend({
     this._super(...arguments);
-    let location = this.get('location');
-    let mapElement = this.get('maps').getMapElement(location);
+    let location = this.location;
```
this.location can probably be inlined in the following line

cspanring commented:
@jenweber @mansona @CodingItWrong I think I addressed all your feedback. Please let me know if there's anything else. Otherwise we could merge and I could apply the diff to 3.3.
